src/test/java/core/basesyntax/service/impl/WriterCsvImplTest.java

```diff
@@ -0,0 +1,83 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.Writer;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class WriterCsvImplTest {
+    private static Writer fileWriter;
+    private static final String OUTPUT_FILE_PATH = "src/test/java/resources/output_database.csv";
+    private static String report;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @BeforeClass
+    public static void beforeClass() {
+        fileWriter = new WriterCsvImpl();
+        report = "fruit,quantity" + System.lineSeparator()
+                + "apple,63" + System.lineSeparator()
+                + "banana,125";
+    }
+
+    @Test
+    public void writeToFile_validFilePathAndReport_Ok() {
+        fileWriter.writeToFile(report, OUTPUT_FILE_PATH);
+        String expected = report;
+        String actual = null;
+        try {
+            actual = Files.readString(Path.of(OUTPUT_FILE_PATH));
+        } catch (IOException e) {
+            throw new RuntimeException("writeToFile_validFilePathAndReport_Ok");
+        }
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void writeToFile_invalidFilePath_notOk() {
+        String wrongOutputFilePath = "src/test/java/resources/2/output_database.csv";
+        expectedException.expect(RuntimeException.class);
+        expectedException.expectMessage("Can't write data to file " + wrongOutputFilePath);
+        fileWriter.writeToFile(report, wrongOutputFilePath);
+    }
+
+    @Test
+    public void writeToFile_emptyFilePath_notOk() {
+        String wrongOutputFilePath = "";
+        expectedException.expect(RuntimeException.class);
+        expectedException.expectMessage("Can't write data to file " + wrongOutputFilePath);
+        fileWriter.writeToFile(report, wrongOutputFilePath);
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void writeToFile_nullFilePath_notOk() {
+        String wrongOutputFilePath = null;
+        fileWriter.writeToFile(report, wrongOutputFilePath);
+    }
+
+    @Test
+    public void writeToFile_emptyReport_Ok() {
+        String report = "";
+        fileWriter.writeToFile(report, OUTPUT_FILE_PATH);
+        String expected = "";
+        String actual = null;
+        try {
+            actual = Files.readString(Path.of(OUTPUT_FILE_PATH));
+        } catch (IOException e) {
+            throw new RuntimeException("writeToFile_validFilePathAndReport_Ok");
+        }
```

Comment:
Maybe we need to move it into separate private method? And call it when you will nee to read file from data?

Reply:
Hmm... Not sure. I am throw different messages to surely know what of tests was wrote wrong.
Prefer mentor to decide.